src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java

```diff
@@ -499,40 +499,79 @@ public static void close(Closeable closeable) {
      * @throws CheckstyleException on failure
      */
     public static URI getUriByFilename(String filename) throws CheckstyleException {
-        // figure out if this is a File or a URL
+        URI uri = getWebOrFilePrefixUri(filename);
+
+        if (uri == null) {
+            uri = getClasspathOrFilepathUri(filename);
+        }
+
+        return uri;
+    }
+
+    /**
+     * Resolves the specified local filename to a URI.
+     *
+     * @param filename name of the file
+     * @return resolved header file URI
+     * @throws CheckstyleException on failure
+     */
+    private static URI getClasspathOrFilepathUri(String filename) throws CheckstyleException {
+        final URI uri;
+        if (new File(filename).exists()) {
+            final File file = new File(filename);
+            uri = file.toURI();
+        }
+        else {
+            uri = getResources(filename);
+        }
+        return uri;
+    }
+
+    /**
+     * Resolves the specified filename containing 'http', 'https', 'ftp',
+     * and 'file' protocols (or any RFC 2396 compliant URL) to a URI.
+     *
+     * @param filename name of the file
+     * @return resolved header file URI or null if URL is malformed or non-existent
+     */
+    public static URI getWebOrFilePrefixUri(String filename) {
```

Comment:
move it above `getClasspathOrFilepathUri` as it appears in code sooner.
Keep in mind - https://github.com/sevntu-checkstyle/methods-distance

Reply:
Done.